tests/jobs/scala/src/main/scala/KerberizedKafkaProducer.scala

```diff
@@ -35,14 +35,28 @@ object KerberizedKafkaProducer {
 
   def sendMessages(topic: String, messages: Array[String])(producer: KafkaProducer[Int, String]): Unit = {
     messages.foreach { str =>
-      val msg = new ProducerRecord[Int, String](topic, str.hashCode, str)
+      val msg = new ProducerRecord[Int, String](topic, str)
       println(s"sending message: $msg")
 
-      Try(producer.send(msg)) match {
-        case Success(_) => println("Message sent")
-        case Failure(ex) => ex.printStackTrace()
+      producer.send(msg, new Callback {
+      override def onCompletion(metadata: RecordMetadata, exception: Exception): Unit = {
+        if (exception != null) {
+          exception.printStackTrace()
+        } else {
+          println(s"Message has been published to (topic_partition@offset): ${metadata.toString}")
+        }
       }
+      })
     }
+    Thread.sleep(3000) // wait a few seconds for metrics update
```

Comment:
let's avoid using `Thread.sleep` and use scheduled executor service for metrics polling and terminating it when metrics appear.

Reply:
oh, this one is a bit tricky. when I troubleshooted the issue with producing to kafka, I observed, that the method `producer.metrics()`, which is invoked right after `send()` method, sometimes returns intermediate results, which means metric has already been added, but has not been updated in the moment of method invocation. `sleep` for a few seconds seems to solve this issue,  but yeah, I need to fine more elegant solution for that

Reply:
It seems like the issue with metrics has been resolved after switching to synchronous send method. I propose to use synchronous method in order to prevent various concurrency issues, that might impact test results as well.